samza-core/src/main/java/org/apache/samza/job/model/JobModel.java

```diff
@@ -143,6 +144,7 @@ private void populateContainerLocalityMappings() {
     return containers;
   }
 
+  @JsonIgnore
```

Comment:
How does adding this fix existing versions? If it doesn't, why ignore it instead of serde-ing it correctly?

Reply:
Discussed offline. 

Currently public fields(maxChangeLogStreamPartitions) and public getter methods (getTaskProcessorLocality, getAllContainerLocality) in JobModel is included in the serialization. Exception was due to that.

Agreed short term fix is to inline the getTaskPartitionMappings method invocation (rather than using JsonIgnore). 

From the JsonJobModelMixin class, it looks like we wanted to serialize only jobModel.config and  jobModel.containers. Will create a JIRA for the long term fix(We can follow similar pattern used in SSP, TaskName json serde and choose a subset of fields to serialize/deserialize).

Reply:
Just to clarify, in addition to getTaskPartitionMappings, getTaskProcessorLocality, getAllContainerLocality and maxChangeLogStreamPartitions are causing the serialization exception. Is that a correct statement?

Reply:
I dont think so. getAllContainerLocality and maxChangeLogStreamPartitions  were present in JobModel from really earlier samza versions(>=0.7). Hence they were not causing problem.